ui/nuxeo-document-layout.js

```diff
@@ -0,0 +1,303 @@
+import '@polymer/polymer/lib/elements/dom-repeat.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { afterNextRender } from '@polymer/polymer/lib/utils/render-status.js';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class';
+import { I18nBehavior } from './nuxeo-i18n-behavior.js';
+
+/**
+ * An element to import and stamp a document layouts.
+ *
+ * Example:
+ *
+ *     <nuxeo-document-layout document='{...}' layout="edit"></nuxeo-document-layout>
+ *
+ * Layouts are stamped by convetion. The href is built based on the document's type (`document.type`)
+ * and the layout's name (`layout`). The layout's href is resolved in relation to either the parent or
+ * this elements `importPath`, so long as they have `importMeta` defined. The base url can be changed via
+ * `hrefBase`, and the whole href pattern can be configured with `hrefTemplate`, The default is:
+ *
+ * `${document.type}/nuxeo-${document.type)-${layout}-layout.html`
+ *
+ * @element nuxeo-document-layout
+ * @memberof Nuxeo
+ * @demo https://nuxeo.github.io/nuxeo-elements/?path=/story/ui-nuxeo-document-layout--default
+ */
+{
+  class DocumentLayout extends mixinBehaviors([I18nBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style>
+          #error {
+            margin-bottom: 8px;
+          }
+
+          .error {
+            border-left: 4px solid var(--paper-input-container-invalid-color, red);
+            color: var(--paper-input-container-invalid-color, red);
+            padding-left: 8px;
+          }
+        </style>
+        <div id="error">
+          <dom-repeat items="[[_errorMessages]]">
+            <template>
+              <span class="error">[[item]]</span>
+            </template>
+          </dom-repeat>
+        </div>
+        <nuxeo-layout
+          id="layout"
+          href="[[_href]]"
+          model="[[_model]]"
+          error="[[i18n('documentLayout.notFound', layout, document.type)]]"
+          on-element-changed="_elementChanged"
+        >
+        </nuxeo-layout>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-layout';
+    }
+
+    static get importMeta() {
+      return import.meta;
+    }
+
+    static get properties() {
+      return {
+        /**
+         * The document for which the layout will be stamped.
+         */
+        document: {
+          type: Object,
+          notify: true,
+        },
+        /**
+         * An id denoting the name of the layout to be stamped.
+         */
+        layout: {
+          type: String,
+          value: 'view',
+          reflectToAttribute: true,
+        },
+        /**
+         * The template to be used to build the layout href, starting with `hrefBase`.
+         */
+        hrefTemplate: {
+          type: String,
+          // eslint-disable-next-line no-template-curly-in-string
+          value: () => '${hrefBase}${document.type}/nuxeo-${document.type}-${layout}-layout.html',
```

Comment:
since we have an `hrefBase`  property I don't think it should be part of the template. It should always be the base anb the template should everything that comes after it